rust/gui-client/src-tauri/src/client/gui/system_tray_menu.rs

```diff
@@ -3,146 +3,164 @@
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
 pub(crate) enum Window {
     About,
     Settings,
 }
 
-#[derive(thiserror::Error, Debug)]
-pub(crate) enum Error {
-    #[error("the system tray menu item ID is not valid")]
-    InvalidId,
-}
-
-const ABOUT_KEY: &str = "/about";
-const SETTINGS_KEY: &str = "/settings";
-const QUIT_KEY: &str = "/quit";
-const QUIT_ACCELERATOR: &str = "Ctrl+Q";
-
-impl FromStr for Event {
-    type Err = Error;
-
-    fn from_str(s: &str) -> Result<Self, Error> {
-        Ok(match s {
-            ABOUT_KEY => Self::ShowWindow(Window::About),
-            "/cancel_sign_in" => Self::CancelSignIn,
-            SETTINGS_KEY => Self::ShowWindow(Window::Settings),
-            "/sign_in" => Self::SignIn,
-            "/sign_out" => Self::SignOut,
-            QUIT_KEY => Self::Quit,
-            s => {
-                let id = s.strip_prefix("/resource/").ok_or(Error::InvalidId)?;
-                Self::Resource { id: id.to_string() }
-            }
-        })
-    }
-}
+const QUIT_TEXT_SIGNED_OUT: &str = "Quit Firezone";
 
 pub(crate) fn signed_in(user_name: &str, resources: &[ResourceDescription]) -> SystemTrayMenu {
     let mut menu = SystemTrayMenu::new()
-        .add_item(
-            CustomMenuItem::new("".to_string(), format!("Signed in as {user_name}")).disabled(),
-        )
-        .add_item(CustomMenuItem::new("/sign_out".to_string(), "Sign out"))
-        .add_native_item(SystemTrayMenuItem::Separator)
-        .add_item(CustomMenuItem::new("".to_string(), "Resources").disabled());
+        .disabled(format!("Signed in as {user_name}"))
+        .item(Event::SignOut, "Sign out")
+        .separator()
+        .disabled("Resources");
 
     for res in resources {
-        let id = res.id();
-        let submenu = SystemTrayMenu::new().add_item(CustomMenuItem::new(
-            format!("/resource/{id}"),
-            res.pastable(),
-        ));
+        let submenu = SystemTrayMenu::new();
+
+        let address_description = res.address_description();
+        let submenu = match Url::parse(address_description) {
+            Ok(url) => submenu.item(Event::Url(url), address_description),
+            Err(_) => submenu.copyable(address_description),
+        };
+
+        let submenu = submenu
+            .separator()
+            .disabled("Resource")
+            .copyable(res.name())
+            .copyable(res.pastable().as_ref());
+
+        let submenu = if let Some(site) = res.sites().first() {
+            // This should roughly match the iChat status indicators used in the macOS Client.
+            // Windows doesn't support colors, and the shapes are needed for colorblind users anyway.
+            // <https://wearecolorblind.com/examples/ichat/>
+            let status = match res.status() {
+                Status::Unknown => "⭕ No activity",
+                Status::Online => "🟢 Gateway connected",
+                Status::Offline => "🟥 All Gateways offline",
```

Comment:
This looks a bit out of place -- what do you think of using the red circle here and just leaving the No activity case empty, or a dash?